### PR TITLE
Use SecRel 4.0

### PR DESCRIPTION
**Pull request by yoomlam**
## What was the problem?

A release candidate for major version 4.0.0 has been released.
https://dsva.slack.com/archives/C03UA9MV1EH/p1675107172862209
> Teams must migrate to v4.0.0 by Feb 14th

Associated tickets or Slack threads:
- #888

## How does this fix it?[^1]

Creates new GH Action workflow to use SecRel 4. Also update libraries to address new SecRel alerts.

TODO: Once it's working consistently, disable SecRel 3 workflow.

## How to test this PR
- Manually run workflow [^secrel]

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: .github/workflows/snyk.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 name: VRO's Snyk scan
+# TODO: delete this once we're using SecRel Modularized Snyk: snyk.yml@v4.0.0
```
Comment (yoomlam) [inline .github/workflows/snyk.yml:2]: TODO is captured in #1005

File: .github/workflows/trivy.yml [hunk 2]
```diff
@@ -1,4 +1,6 @@
 name: Trivy scan
+# TODO: delete this once we're using SecRel's Modularized Aqua: aqua.yml@v4.0.0
```
Comment (yoomlam) [inline .github/workflows/trivy.yml:2]: TODO is captured in #1005

File: gradle.properties [hunk 3]
```diff
@@ -39,7 +39,7 @@ shellcheck_plugin_version=1.4.6
 shellcheck_tool_version=v0.7.1
 spotless_plugin_version=5.14.1
 spring_security_version=5.7.5
-spring_boot_version=2.7.4
+spring_boot_version=2.7.6
```
Comment (yoomlam) [inline gradle.properties:42]: To address Snyk alerts

File: .github/workflows/secrel4prs.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-name: "SecRel check for PRs"
+name: "SecRel 4 pipeline"
```
Comment (yoomlam) [inline .github/workflows/secrel4prs.yml:1]: Before merging, I'll rename this file to secrel4.yml. I'm keeping the current name so I can run this workflow manually for testing purposes. 
Ticket #1005 will create modular SecRel workflows for PRs.

**Review by sethdarragile6 — APPROVED** [review]
LGTM!

**Pull request merged**